File: 009.py
#!/usr/bin/env python

"""Find the product abc where a, b, c are a Pythagorean triplet for which a + b + c = 1000."""

TRIPLET_SUM = 1000

def is_pythagorean_triplet(a, b, c):
    """Determine if a, b, c are a Pythagorean triplet."""
    return (a ** 2 + b ** 2) == (c ** 2)

def find_abc():
    for a in xrange(1, 332 + 1):
        for b in xrange(2, 499 + 1):
            if a < b:
                c = TRIPLET_SUM - a - b
                if b < c:
                    if is_pythagorean_triplet(a, b, c):
                        return (a, b, c)

a, b, c = find_abc()
print a * b * c

File: 009.swift
#!/usr/bin/xcrun swift

import Foundation

// Find the product abc where a, b, c are a Pythagorean triplet for which a + b + c = 1000.

let tripletSum = 1000

func isPythagoreanTriplet(a: Int, b: Int, c: Int) -> Bool {
    // Determine if a, b, c are a Pythagorean triplet.
    return (pow(Double(a), 2.0) + pow(Double(b), 2.0)) == pow(Double(c), 2.0)
}

func findABC() -> (Int, Int, Int)? {
    for a in 1...332 {
        for b in 2...499 {
            if a < b {
                let c = tripletSum - a - b
                if b < c {
                    if isPythagoreanTriplet(a, b, c) {
                        return (a, b, c)
                    }
                }
            }
        }
    }
    return nil
}

if let (a, b, c) = findABC() {
    println(a * b * c)
}
